prov/util/src/util_mem_monitor.c

```diff
@@ -277,12 +297,21 @@ void ofi_monitors_init(void)
 
 void ofi_monitors_cleanup(void)
 {
-	uffd_monitor->cleanup(uffd_monitor);
-	memhooks_monitor->cleanup(memhooks_monitor);
-	cuda_monitor->cleanup(cuda_monitor);
-	rocr_monitor->cleanup(rocr_monitor);
-	ze_monitor->cleanup(ze_monitor);
-	import_monitor->cleanup(import_monitor);
+	struct ofi_mem_monitor *monitors[] = {
+		uffd_monitor,
+		memhooks_monitor,
+		cuda_monitor,
+		rocr_monitor,
+		ze_monitor,
+		import_monitor,
+	};
```

Comment:
I know it's an existing issue: the monitors being cleaned up doesn't match those are initialized.  How about defining the monitors array as file scope static and use the same array for both initialization and cleanup?

Reply:
I have implemented the later, ie, save the list at init time.  This change also calls cleanup on all the monitors (some of which did not have cleanup() processing previously).

Reply:
@muttormark I mean the array you defined here, why can't it be moved outside the functions, and adding a "static" keyword to limit the scope? This way the same array is shared by init and cleanup.